src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -121,27 +131,93 @@ public String getJavaExpressionWithAppend() {
   }
 
   /**
+   * Holder class for results and result counts from {@link #evaluateColumn}.
+   */
+  public static class EvaluatorResult {
+    private final ReadableColumn results;
+    private final Column<?> resultCounts;
+
+    protected EvaluatorResult(
+        @Nonnull final ReadableColumn results, @Nonnull final Column<?> resultCounts) {
+      this.results = results;
+      this.resultCounts = resultCounts;
+    }
+    /**
+     * @return a {@link ReadableColumn} containing results from {@link #evaluateColumn}
+     */
+    public ReadableColumn getResults() {
```

Comment:
After taking your suggestion to return just `List<Column>` from this method, I ended up reverting to a similar interface as before. It's just too awkward to construct the Cartesian product with a `List<Column>` instead of a `TupleBuffer` (keeping in mind that these sequences of columns are "vertical", not "horizontal", since the `TupleBuffer` schema has only one field). The additional bookkeeping adds enough complexity to outweigh the simplification of the interface. Hopefully we can find a solution that simplifies both the interface definition and the client code...